src/Marten/IQuerySession.cs

```diff
@@ -317,5 +317,28 @@ public interface IQuerySession : IDisposable
         /// See: https://www.postgresql.org/docs/10/static/textsearch-controls.html#TEXTSEARCH-PARSING-QUERIES
         /// </remarks>
         Task<IReadOnlyList<TDoc>> PhraseSearchAsync<TDoc>(string searchTerm, string regConfig = FullTextIndex.DefaultRegConfig, CancellationToken token = default);
+        
+        /// <summary>
+        /// Performs a full text search against <typeparamref name="TDoc"/> using the 'websearch_to_tsquery' search function
+        /// </summary>
+        /// <param name="searchTerm">The text to search for.  Uses an alternative syntax to the other search functions, similar to the one used by web search engines</param>
+        /// <param name="regConfig">The dictionary config passed to the 'websearch_to_tsquery' function, must match the config parameter used by <seealso cref="DocumentMapping.AddFullTextIndex(string)"/></param>
+        /// <remarks>
+        /// Supported from Postgres 11
+        /// See: https://www.postgresql.org/docs/11/static/textsearch-controls.html#TEXTSEARCH-PARSING-QUERIES
+        /// </remarks>
+        IReadOnlyList<TDoc> WebSearch<TDoc>(string searchTerm, string regConfig = FullTextIndex.DefaultRegConfig);

```

Comment:
Suggestion: `WebSearch` as a name looks a bit ambiguous, change it to `WebStyleSearch` or `WebStyleTextSearch`? Thoughts?

Reply:
I had actually thought the same, and independently thought of `WebStyleSearch'! I just used `WebSearch` as it seemed to confirm with the others, but I thought it might come up in review :smile: - will change it later today.